doc/function_types.md

```diff
@@ -841,6 +841,47 @@ int main()
 
 </details>
 
+
+## Multi-objective functions
+
+A multi-objective function is a set of functions or objectives that are
+simultaneously optimized to reach an optima.
+
+In order to optimize a multi-objective function with ensmallen, a class
+implementing the API below is required.
+
+<details open>
+<summary>Click to collapse/expand example code.
+</summary>
+
+```c++
+template<typename MatType = arma::mat>
+class MultiObjectiveFunctionType
```

Comment:
To follow up on @zoq's [comment](https://github.com/mlpack/ensmallen/pull/149#discussion_r392527968), my understanding is that it's sufficient to simply pass a `std::tuple<ArbitraryFunctionType1, ArbitraryFunctionType2, ...>` to a multi-objective optimizer.  That's really nice because it's very low-overhead, and we can simply supply the example that @zoq linked to here: https://github.com/mlpack/ensmallen/pull/149/files#diff-b0e728716d6532048249f3bcf1562e03R1602

I think it might be clearer if we replaced this example with another example that just strings together a couple arbitrary functions:

```
In order to optimize a multi-objective function with ensmallen, a `std::tuple<>`
containing multiple `ArbitraryFunctionType`s ([see here](#arbitrary-functions))
should be passed to a multi-objective optimizer's `Optimize()` function.

An example below simultaneously optimizes the generalized Rosenbrock function
in 6 dimensions and the Wood function using [NSGA2](#nsga2).

<details open>
<summary>Click to collapse/expand example code.
</summary>

GeneralizedRosenbrockFunction rf(6);
WoodFunction wf;
std::tuple<GeneralizedRosenbrockFunction, WoodFunction> objectives(rf, wf);

// Create an initial point (a random point in 6 dimensions).
arma::mat initialPoint(6, 1, arma::fill::randu);

NSGA2 optimizer;
std::vector<arma::mat> bestFront = optimizer.Optimize(initialPoint, coordinates);
```

I probably got the anchor links wrong there though, and I didn't test the code (I think it will work?).  It's just an idea---let me know what you think. My hope is just to provide the simplest possible documentation to users, so that they see that all they need is a `std::tuple<>` of arbitrary functions and then they think "oh great, this is easy!". Hopefully, at least. :)